ReactiveCocoa/Swift/TupleExtensions.swift

```diff
@@ -40,3 +40,125 @@ internal func repack<A, B, C, D, E, F, G, H, I>(t: (A, B, C, D, E, F, G, H), val
 internal func repack<A, B, C, D, E, F, G, H, I, J>(t: (A, B, C, D, E, F, G, H, I), value: J) -> (A, B, C, D, E, F, G, H, I, J) {
 	return (t.0, t.1, t.2, t.3, t.4, t.5, t.6, t.7, t.8, value)
 }
+
+
+// This will extend 'indexable' collections that can use an integer Index, 
+/// like Arrays to allow some conviences tuple implementations
+public extension SequenceType where Self:Indexable, Self.Index : IntegerLiteralConvertible {
+	
+	private func _get_element<T>(index : Index) -> T {
+		let x = self[index]
+		let t = x as? T
+		precondition(t != nil, "\(x) is not of type \(T.self) at index \(index) of \(Self.self)")
+		return t!
+	}
+	
+	public func toTuple<A,B>() -> (A,B) {
+		return
+			(self._get_element(0),
```

Comment:
That's fine.  The typical use is that SequenceType is of type Any, and we want to covert the Array<Any> back into a type specific tuple.